### DISCUSSION TITLE
CP/M strangeness when starting games on the H: drive

### DISCUSSION THREAD

**Discussion by Jimbob-B** [ExtremeElectronics/RC2040#14] [Q&A] [answered]
Not sure if it's me being stupid or an emulation bug but I'm seeing some strange behaviour when starting the Infocom games on drive H: using the SIO .img file. Listing the contents of this drive shows double entries for the files for some reason but all the other drives are fine - so I see two ZORK2.COM or HITCH.COM etc. Starting the games works but for most of the games typing something in resets the game and starts it again. The anomoly is Zork 3 as this just auto repeats the first message for the game.

For Hitchikers Guide I downloaded a scanned instruction PDF and it has a help section to start the game. Typing this into the game results in nothing being recognised and the game resets. Typing QUIT to exit the game isn't recognised and again it resets to the start.

This seems to happen with both my own compiled version form the Make Pi Pico board or the downloaded UF2 file for github.

I'm using either picoterm or minicom on Mac OS Catalina with a bog standard micro USB cable to the Pico. Also tried the ACIA serial interface option in rc2040.ini and the same behaviour. Any ideas? Not really sure how or if I can debug what is happening ...

**Comment by ExtremeElectronics** [comment]
The Infocom games on H drive are borked. 
I realised this after the last release and haven't had chance to re do the images.

I have working replacements for Zork1 and HHGTTG but the rest are for a different computer system as far as I know.

**Reply by Jimbob-B** [reply]
Thanks and gosh that was quick! I assumed it was just me doing something wrong! Tinkering with terminal settings didn't help and then I wondered if it was a bug somewhere.

I just wanted to see what all the fuss was about with Infocom games and this was definitely a nice retro way way to see.

**Comment by ExtremeElectronics** [comment]
Try these 
[hitchhiker.zip](https://github.com/ExtremeElectronics/RC2040/files/8681606/hitchhiker.zip)

and 
[ZORK1.zip](https://github.com/ExtremeElectronics/RC2040/files/8681608/ZORK1.zip)

You can either put them into the images with CPM manager. 
or using https://rc2014.co.uk/filepackage/ 

and cut (from the web site) and paste into CPM (it automatically runs a:download.com ) 
ensure you have the serial delays in your terminal program. 

Some more details at https://www.extremeelectronics.co.uk/the-rc2040/ still incomplete.

more details here https://www.extremeelectronics.co.uk/author/eeadmin/

**Comment by Jimbob-B** [comment]
Thank you :-) Downloading now after using the handy web site. Would help if I didn't paste the same download twice :-/

Reminds me a bit of the wait while loading ZX Spectrum games but without the colour loading screen and tones.

**Comment by ExtremeElectronics** [comment, marked as answer]
New images posted in SD contents with only the working games. in g: and i:
